doc/contributing.md

```diff
@@ -16,37 +16,208 @@ and we encourage all to read it carefully.
+* Contribute code! This can be code that you already have and it doesn't need to be
+  perfect! We will help you clean things up, test it, etc.
+* Help triage issues, or give a "thumbs up" on issues that others reported which are
+  relevant to you.
+* Participate and answer questions on the [PyGMT forum Q&A](https://forum.generic-mapping-tools.org/c/questions/pygmt-q-a/11).
+* Participate in discussions at the quarterly PyGMT Community Meetings, which are
+  announced on the [forum governance page](https://forum.generic-mapping-tools.org/c/governance/9).
+* Spread the word about PyGMT or star the project!
 
+## Providing Feedback
 
-## Getting Help
+### Reporting a Bug
 
-Discussion often happens on GitHub issues and pull requests.
-In addition, there is a
+* Find the [*Issues*](https://github.com/GenericMappingTools/pygmt/issues) tab on the
+top of the GitHub repository and click *New Issue*.
+* Click on *Get started* next to *Bug report*.
+* **Please try to fill out the template with as much detail as you can**.
+* After submitting your bug report, try to answer any follow up questions about the bug
+  as best as you can.
+
+### Submitting a Feature Request
+
+* Find the [*Issues*](https://github.com/GenericMappingTools/pygmt/issues) tab on the
+  top of the GitHub repository and click *New Issue*.
+* Click on *Get started* next to *Feature request*.
+* **Please try to fill out the template with as much detail as you can**.
+* After submitting your feature request, try to answer any follow up questions as best
+  as you can.
+
+### Submitting General Comments/Questions
+
+There are several pages on the [Community Forum](https://forum.generic-mapping-tools.org/)
+where you can submit general comments and/or questions:
+
+* For questions about using PyGMT, select *New Topic* from the
+  [PyGMT Q&A Page](https://forum.generic-mapping-tools.org/c/questions/pygmt-q-a/11).
+* For general comments, select *New Topic* from the
+  [Lounge Page](https://forum.generic-mapping-tools.org/c/lounge/6).
+* To share your work, select *New Topic* from the
+  [Showcase Page](https://forum.generic-mapping-tools.org/c/Sow-your-nice-example-script/10).
+
+## General Guidelines
+
+### Resources for New Contributors
+
+Please take a look at these resources to learn about Git and pull requests (don't
+hesitate to [ask questions](#getting-help)):
+
+* [How to Contribute to Open Source](https://opensource.guide/how-to-contribute/).
+* Aaron Meurer's [tutorial on the git workflow](http://www.asmeurer.com/git-workflow/).
+* [How to Contribute to an Open Source Project on GitHub](https://egghead.io/courses/how-to-contribute-to-an-open-source-project-on-github).
+
+### Getting Help
+
+Discussion often happens on GitHub issues and pull requests. In addition, there is a
 [Discourse forum](https://forum.generic-mapping-tools.org/c/questions/pygmt-q-a) for
 the project where you can ask questions.
 
+### Setting up your Environment
```

Comment:
If I am a new contributor, I can just read the first few sections **Ways to Contribute**,  **Providing Feedback**, **Resources for New Contributors**, **Getting Help**, **Pull Request Workflow** and they are now easy to follow since you have polished and organize them so well. But I don't need **Setting up your Environment** until I want to revise the documentation or write codes locally, right? So it would be better to put it within **Contributing Documentation** or **Contributing Code**. I don't know how others feel about this subsection.

Reply:
I agree that **Setting up your Environment** might not be appropriate for **General Guideline**, but don't like it under **Contributing Documentation** or **Contributing Code** because it can be relevant for both. What if we instead just move it to its own second level heading after **General Guidelines** and before **Contributing Documentation** with a short paragraph at the start explaining when it is needed (i.e., for editing documentation locally and adding code)?

Reply:
> but don't like it under Contributing Documentation or Contributing Code because it can be relevant for both. 

I agree, that's a good point.

> What if we instead just move it to its own second level heading after **General Guidelines** and before **Contributing Documentation** with a short paragraph at the start explaining when it is needed (i.e., for editing documentation locally and adding code)?

Sounds good. BTW, we could create a second level section for all the common things (like **Setting up your Environment**) for  **Contributing Documentation** and  **Contributing Code** in the future if we have more than one.